src/installer/tests/HostActivation.Tests/NativeHosting/GetFunctionPointer.cs

```diff
@@ -0,0 +1,287 @@
+
+using Microsoft.DotNet.Cli.Build.Framework;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public partial class GetFunctionPointer : IClassFixture<GetFunctionPointer.SharedTestState>
+    {
+        private const string ComponentGetFunctionPointerArg = "component_get_function_pointer";
+        private const string AppGetFunctionPointerArg = "app_get_function_pointer";
+
+        private readonly SharedTestState sharedState;
+
+        public GetFunctionPointer(SharedTestState sharedTestState)
+        {
+            sharedState = sharedTestState;
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        public void CallDelegateOnApplicationContext(bool validType, bool validMethod)
+        {
+            var appProject = sharedState.ApplicationFixture.TestProject;
+            string[] args =
+            {
+                AppGetFunctionPointerArg,
+                sharedState.HostFxrPath,
+                appProject.AppDll,
+                validType ? sharedState.FunctionPointerTypeName : $"Component.BadType, {appProject.AssemblyName}",
+                validMethod ? sharedState.FunctionPointerEntryPoint1 : "BadMethod",
+            };
+            CommandResult result = sharedState.CreateNativeHostCommand(args, sharedState.DotNetRoot)
+                .Execute();
+
+            result.Should()
+                .InitializeContextForApp(appProject.AppDll);
+
+            if (validType && validMethod)
+            {
+                result.Should().Pass()
+                    .And.ExecuteFunctionPointer(sharedState.FunctionPointerEntryPoint1, 1, 1);
+            }
+            else
+            {
+                result.Should().Fail();
+            }
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        public void CallDelegateOnComponentContext(bool validType, bool validMethod)
+        {
+            var componentProject = sharedState.ComponentWithNoDependenciesFixture.TestProject;
+            string[] args =
+            {
+                ComponentGetFunctionPointerArg,
+                sharedState.HostFxrPath,
+                componentProject.RuntimeConfigJson,
+                validType ? sharedState.ComponentTypeName : $"Component.BadType, {componentProject.AssemblyName}",
+                validMethod ? sharedState.ComponentEntryPoint1 : "BadMethod",
+            };
+            CommandResult result = sharedState.CreateNativeHostCommand(args, sharedState.DotNetRoot)
+                .Execute();
+
+            result.Should()
+                .InitializeContextForConfig(componentProject.RuntimeConfigJson);
+
+            result.Should().Fail();
```

Comment:
Please add a comment why it always fails (the component type won't resolve in the default ALC)